### DISCUSSION TITLE
Cheaper bot hosting: VPS

### DISCUSSION THREAD

**Discussion by DudeShift** [Mateo-Wallace/Discord-DumBot#20] [General]
In your docs, you said you were looking for a way to run the bot for free. Well I recommend using a VPS and docker. Although a VPS isn't free, they can be far less expensive then Heroku. From [Heroku dyno types]((https://devcenter.heroku.com/articles/dyno-types) the Basic $7/mo is 512MB ram and 1 cpu share. Checking around for deals (see [lowendtalk](https://lowendtalk.com/) or [lowendbox](https://lowendbox.com/)) at say RackNerd, you can get a [1GB ram and 1 vCPU core for $11/year](https://my.racknerd.com/cart.php?a=confproduct&i=4) (yes, per year). There are plenty of different offers from RackNerd (if you want to run mulitple bots / projects) or different VPS providers.

From my 2 year experience using RackNerd, they usually have offers not shown on their own website but found from lowendtalk/box pages. For example their [Black Friday 2022 offers](https://my.racknerd.com/index.php?rp=/store/black-friday-2022) are still available, and here are their current [Easter 2023 offers](https://lowendtalk.com/discussion/185231/easter-deals-new-inventory-new-deals-excellent-pricing-fantastic-support). But one would have to keep an eye for other VPS providers deals like for example [HostVDS](https://lowendtalk.com/discussion/182860/no-discount-just-0-99-mo-all-the-time-for-1-vcpu-1gb-ram-10gb-nvme-and-hourly-billing).

Getting a VPS to a working bot takes a few steps. Most providers offer a VNC (accessing the VPS from a web browser) or you can login to it via SSH something like Putty or MobaXterm. But once logined in its basically your own server to install what you want. I recommend using docker to run your applications, projects, bots in as its easy to have multiple projects run in the background.

To run this music bot in a docker container I made a pull request with the Dockerfile needed and the steps to make the docker container.

**Comment by Mateo-Wallace** [comment]
Adding this link here to easily find in the future:
[Instructions for how to setup Docker for a VPS](https://github.com/Mateo-Wallace/MP2-Discord-DumBot-V2/pull/19#issue-1659042060)

**Comment by Mateo-Wallace** [comment]
Hi DudeShift, 

Thanks so much for taking a look at the project and making a contribution. I really like the idea and love the pricepoint. I'm going to keep in the docs that I'm still looking for a free version but this seems like a much better solution than Heroku. 

The only thing I don't like about it is it seems like you have to input your stuff through the command line. Nothing wrong with that for people who are familiar with the command line but DumBot's main purpose is for people who don't know how to use code at all. Thats why I went with heroku because its as simple as connect the repo and boom good to go. No command line needed or cloning needed. 

I might make a developer section to add this method in there. 

Thanks so much!

-Mateo
